internal/cmd/gocdk/_static/init/Dockerfile

```diff
@@ -0,0 +1,17 @@
+# DO NOT REMOVE! The following line informs the Go CDK of the Docker image's
```

Comment:
I have the same pondering of "is the distinction between the CLI and the overall project useful to the user?" mentioned elsewhere. While it's important for us to be precise in internal documentation or reference, I think the user will think of the "Go CDK" as a gestalt. We should keep it consistent across each of the docs, so I want us to decide once and then I'm happy to rewrite the files all together as needed.

The version information bit is trying to warn them not to put a tag on it, but I agree it's not clear and probably better to do without. WDYT?

/cc @vangent so as not to miss the conversation

Reply:
`gocdk` seems more appropriate here to me, as it is more specific.

Side note: can't we figure this out dynamically?

Reply:
I don't know that I agree that we should keep consistency between "the Go CDK" and "gocdk" usage in these docs. Sometimes we mean the whole project, and sometimes we mean only the cli. Here, it's only the cli that will use it.

@vangent, if the image name is the same as the project dir name, we presumably can, but it also is nice for it to be explicit. Being explicit, it show them where they would change it if they wanted to.

Reply:
> it show them where they would change it if they wanted to.

But, IIUC, just changing it at this comment would not work. The same variable is used several other times in this same file.

Reply:
ah, you're right about that. We'd need to afford an option for them to explicitly set it if they want it to be difference. Could [eventually/not in this PR] be a flag on init?

Reply:
@clausti:

> I don't know that I agree that we should keep consistency between "the Go CDK" and "gocdk" usage in these docs. Sometimes we mean the whole project, and sometimes we mean only the cli. Here, it's only the cli that will use it.

Sorry for imprecision: I mean to say that there's a few different instances in this PR where it's talking about CLI (`gocdk`)-specific things, and I wanted to perform the edits at the same time. Sounds like the agreement is for `gocdk`, so changed!

@vangent: 

> Side note: can't we figure this out dynamically?

Out of scope for this PR, but no. There is nothing in the Dockerfile format (intentionally) that specifies what the name/tag for the build should be. That's why `docker build` commands typically take the form `docker build -t myimage .`. This information needs to live somewhere, and people usually leave comments in their Dockerfiles anyway with this info.

> But, IIUC, just changing it at this comment would not work. The same variable is used several other times in this same file.

Your understanding is incorrect. The non-comment usages refer to where the binary is placed inside the container, and as long as they are self-consistent, it's largely hidden from the user of the Docker image. Changing the magic comment will indicate to the CLI what tag to build the image with.

Reply:
We could figure it out dynamically if that means "read the directory name and always use that", even if we can't read it out of the Dockerfile, but I agree that's not better than the comment.

Reply:
Directory name isn't a good proxy for this information since Docker image names can (and frequently do) contain slashes. And besides, it would be surprising if someone checked out their source repo with a slightly different directory name and broke their deploy process.

I really feel like this out of scope for this PR at this point, what's being discussed here is a wholly separate change. Does this block submission?

Reply:
Not for me; I approved yesterday.

Reply:
No, it doesn't block submission. If you want to merge it with @vangent's green I won't be offended. If you want both && me to understand the code I'm approving, I'm not done yet. :/

Reply:
> No, it doesn't block submission. If you want to merge it with @vangent's green I won't be offended. If you want both && me to understand the code I'm approving, I'm not done yet. :/

That's fine, I can wait on that. My question was whether dynamically determining Docker image name blocks this PR.

I was confused by the shift in the conversation from the wording in documentation (which I've changed now) to a conversation about how the Docker image is named and changing that. Changing that behavior would require logic changes beyond just `gocdk init`.